lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -77,6 +83,9 @@ def rabbitmq_settings
           :automatic_recovery => @automatic_recovery,
           :pass => @password ? @password.value : "guest",
         }
+
+        s[:timeout] = @connection_timeout if @connection_timeout
+        s[:heartbeat] = @heartbeat if @heartbeat
```

Comment:
The way I read this, this is always called because all numbers are truthy in Ruby.

Since (at the time of my review) the `:default => 0` is used for heartbeat, and it's validated as number, you'll never have `@heartbeat` be anything but a truthy value. Some examples:

```
=> nil
>> puts "whoa" if 0
whoa
=> nil
>> puts "whoa" if -1
whoa
=> nil
>> puts "whoa" if 1
whoa
```

In Ruby, 0 is truthy (all fixnums are! hehe). Unlike say, python, where `False` is really just 0 and coerces as to a number. Computerrrrrsssss

```python
# in python...
>>> if 0:
...   print "whoa"
... else:
...   print "0 is false in python"
...
0 is false in python

>>> False+0
0
```